### PR TITLE
Implement C API for datatable

### LINKED ISSUE #1382: Exposing C APIs
As I understand, the C APIs are not exposed.

I think there are at least two steps to expose the C APIs:

1. installing the header files, like `numpy/core/include`

2. providing utility functions, like `numpy.get_include()`

### PR DESCRIPTION
**Pull request by st-pasha**
This PR adds external C API to datatable. The user of this API will have to include the file "datatable.h" located in directory "datatable/include/". This file will also be included in `datatable` source and binary distributions.

The following API functions are currently implemented:

- `size_t DtABIVersion();`
  Check the ABI version of the current datatable package. This method allows you to verify that the runtime ABI version of datatable matches your compile-time expectations.

- `int DtFrame_Check(PyObject* ob)`
  Check whether the python object is a datatable Frame.

- `size_t DtFrame_NRows(PyObject* pydt)`, `size_t DtFrame_NColumns(PyObject* pydt)`
  Get the shape of the Frame.

- `int DtFrame_ColumnStype(PyObject* pydt, size_t i)`
  Get the stype of column `i`.

- `PyObject* DtFrame_ColumnRowindex(PyObject* pydt, size_t i)`  
  Get the Rowindex of column `i`.

- `const void* DtFrame_ColumnDataR(PyObject* pydt, size_t i)`, `void* DtFrame_ColumnDataW(PyObject* pydt, size_t i)`, `const char* DtFrame_ColumnStringDataR(PyObject* pydt, size_t i)`
  Get the data pointer(s) of column `i`.

- `int DtRowindex_Check(PyObject* ob)`
  Check whether the python object is a datatable Rowindex.

- `int DtRowindex_Type(PyObject* pyri)`
  Find the type of the Rowindex.

- `size_t DtRowindex_Size(PyObject* pyri)`
  Get the number of elements in the Rowindex.

- `int DtRowindex_UnpackSlice(PyObject* pyri, size_t* start, size_t* length, size_t* step)`
  Get fields of a slice Rowindex.

- `const void* DtRowindex_ArrayData(PyObject* pyri)`
  Get data of an array Rowindex.

Closes #1382

### CODE REVIEW

File: datatable/include/datatable.h [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+/**
+ * Return a pointer to the string data buffer of a column `i` in Frame `pydt`.
+ * If the column is not of string type (STR32 or STR64), a null pointer will
+ * be returned and an error set.
+ *
+ * The returned pointer is a borrowed reference. It may be invalidated during
+ * the subsequent calls to datatable.
+ */
+const char* DtFrame_ColumnStringDataR(PyObject* pydt, size_t i);
```
Comment (thirdwing) [inline datatable/include/datatable.h]: I just have a very simple question. Why there is no `DtFrame_ColumnStringDataW`?

Comment (st-pasha) [reply datatable/include/datatable.h]: Because there aren't any useful operations where you could modify string data in-place.

**Comment by thirdwing** [comment]
Can we have a new release including this PR? Maybe 0.7.1?

**Comment by st-pasha** [comment]
@thirdwing We can make a new release closer to the end of the month. For now you can use nightly builds published on S3.

**Pull request merged**